proposals/0000-reduce-fragmentation.md

```diff
@@ -0,0 +1,837 @@
+
+In some cases, a particular feature of the base Node interface may not apply to one of its child interfaces; in that case, the inheriting node may return null or throw an exception, depending on circumstances.
+
+### Methods
+
+In addition to the properties below, Node inherits methods from its parent, EventTarget.
+
+- [ ] `Node.contains()`. Returns true or false value indicating whether or not a node is a descendant of the calling node.
+- [ ] `Node.getRootNode()`. Returns the context object's root which optionally includes the shadow root if it is available..
+
+## Element
+
+Element is the most general base class from which all element objects (i.e. objects that represent elements) in a Document inherit. It only has methods and properties common to all kinds of elements. More specific classes inherit from Element.
+
+### Properties
+
+Element inherits properties from its parent interface, Node, and by extension that interface's parent, EventTarget.
+
+- [ ] `Element.clientHeight` [Read only]. Returns a number representing the inner height of the element.
+- [ ] `Element.clientLeft` [Read only]. Returns a number representing the width of the left border of the element.
+- [ ] `Element.clientTop` [Read only]. Returns a number representing the width of the top border of the element.
+- [ ] `Element.clientWidth` [Read only]. Returns a number representing the inner width of the element.
+- [ ] `Element.id` [Read only]. Is a DOMString representing the id of the element.
+- [ ] `Element.scrollHeight` [Read only]. Returns a number representing the scroll view height of an element.
+- [ ] `Element.scrollLeft`. Is a number representing the left scroll offset of the element.
+- [ ] `Element.scrollLeftMax` [Read only]. Returns a number representing the maximum left scroll offset possible for the element.
+- [ ] `Element.scrollTop`. A number representing number of pixels the top of the element is scrolled vertically.
+- [ ] `Element.scrollTopMax` [Read only]. Returns a number representing the maximum top scroll offset possible for the element.
+- [ ] `Element.scrollWidth` [Read only]. Returns a number representing the scroll view width of the element.
+
+### Methods
+
+Element inherits methods from its parents Node, and its own parent, EventTarget.
+
+- [ ] `Element.computedStyleMap()`. Returns a StylePropertyMapReadOnly interface which provides a read-only representation of a CSS declaration block that is an alternative to CSSStyleDeclaration.
+- [ ] `Element.getBoundingClientRect()` **P1**. Returns the size of an element and its position relative to the viewport.
+- [ ] `Element.scroll()`. Scrolls to a particular set of coordinates inside a given element.
+- [ ] `Element.scrollBy()`. Scrolls an element by the given amount.
+- [ ] `Element.scrollIntoView()`. Scrolls the page until the element gets into the view.
+- [ ] `Element.scrollTo()`. Scrolls to a particular set of coordinates inside a given element.
```

Comment:
These properties are often used in event handling.

Reply:
There have been [some early internal discussions](https://fburl.com/d1pak0i8) about a platform-agnostic React Component Instance API that makes the "best practice" imperative APIs available while abstracting away most of the more intrusive tree traversal APIs (which violation encapsulation of React components).

A lot of the APIs you've identified here as typically valuable in event handling, were also included in that.

Reply:
Need to add:

* `getClientRects()`
* `compareDocumentPosition()`

Will keep `contains()` and `getBoundingClientRect()` because they are used far more frequently than the above, and easy to derive. And will not add the non-standard APIs in that proposal.